### ISSUE TITLE
[Feature] Deprecate `batch_size` in favor of `gradient accumulation steps`

### ISSUE THREAD

**Issue by NanoCode012** [axolotl-ai-cloud/axolotl#121]
Reason: Easier for calculation

- [x] Replace use of batch size with gradient
- [ ] Replace in doc
- [x] Check duplicate config in validate_config
- [x] Replace all configs/examples

**Comment by winglian** [comment]
sorry, didn't see this. temp fix for now https://github.com/OpenAccess-AI-Collective/axolotl/pull/123

**Comment by NanoCode012** [comment]
I think your solution is quite nice! I however just noticed something

```py
gas = cfg.batch_size // cfg.micro_batch_size
```

Should we have also divided this by # gpu?

**Comment by winglian** [comment]
that gets handled here:
https://github.com/OpenAccess-AI-Collective/axolotl/blob/main/scripts/finetune.py#L162-L166

**Comment by NanoCode012** [comment]
> that gets handled here:
> https://github.com/OpenAccess-AI-Collective/axolotl/blob/main/scripts/finetune.py#L162-L166

I see now!
